app/models/search/collection_indexer.rb

```diff
@@ -0,0 +1,102 @@
+class CollectionIndexer < Indexer
+  def self.klass
+    "Collection"
+  end
+
+  def self.mapping
+    {
+      "collection" => {
+        properties: {
+          title: {
+            type: "text",
+            analyzer: "collection_title_analyzer",
+            fields: {
+              exact: {
+                type:     "text",
+                analyzer: "exact_collection_analyzer"
+              },
+              keyword: {
+                type: "keyword",
+                normalizer: "keyword_lowercase"
+              }
+            }
+          },
+          challenge_type: {            
+            type: "keyword",
+            null_value: "NULL" 
+          },
+          name: { type: "text", analyzer: "simple" },
+          description: { type: "text", analyzer: "simple" },
+          collection_type: { type: "keyword" },
+          created_at: { type: "date" }
+        }
+      }
+    }
+  end
+
+  def self.settings
+    {
+      analysis: {
+        analyzer: {
+          collection_title_analyzer: {
+            type: "custom",
+            tokenizer: "standard",
+            filter: [
+              "lowercase"
+            ]
+          },
+          exact_collection_analyzer: {
+            type: "custom",
+            tokenizer: "keyword",
+            filter: [
+              "lowercase"
+            ]
+          }
+        },
+        normalizer: {
+          keyword_lowercase: {
+            type: "custom",
+            filter: ["lowercase"]
+          }
+        }
+      }
+    }
+  end
+
+  def document(object)
+    object.as_json(
+      root: false,
+      only: [
+        :id, :name, :title, :description, :parent_id, :collection_type, :created_at
+      ]
+    ).merge(
+      closed: object.closed?,
+      unrevealed: object.unrevealed?,
+      anonymous: object.anonymous?,
+      owner_ids: object.all_owners.pluck(:user_id),
+      moderated: object.moderated?,
+      moderator_ids: object.all_moderators.pluck(:user_id),
+      maintainer_ids: object.maintainers.pluck(:user_id),
+      challenge_type: object.challenge_type,
+      signup_open: object.challenge&.signup_open,
+      signups_open_at: object.challenge&.signups_open_at,
+      signups_close_at: object.challenge&.signups_close_at,
+      assignments_due_at: object.challenge&.assignments_due_at,
+      works_reveal_at: object.challenge&.works_reveal_at,
+      authors_reveal_at: object.challenge&.authors_reveal_at,
+      general_works_count: object.all_approved_works.count,
+      public_works_count: object.all_approved_works.where(restricted: false).count,
+      general_bookmarked_items_count: get_bookmarked_items_count(object), 
+      public_bookmarked_items_count: get_bookmarked_items_count(object, true)
+    )
+  end
+
+  def get_bookmarked_items_count(collection, is_public = false)
```

Comment:
Oh, duh. I was so busy being excited by the scopes that I forgot to check the end result 🤦

Reply:
I'm not quite sure how to count the bookmarkables. Would you mind providing just a bit more context here?

Reply:
A work can have multiple bookmarks associated with it, and a collection can contain multiple of those bookmarks. So counting the number of bookmarks associated with a collection won't necessarily match the number of bookmarked works associated with the same collection. The same is true of series and external works.

To count the bookmarkables associated with a collection, you need to count the number of distinct works, external works, and series that have at least one bookmark in the collection (or in one of the collection's children). You also need to filter by work/external work/series/bookmark visibility, of course.

(`Collection#all_bookmarked_items_count` counts bookmarkables. What makes it unsuitable for calling here is that it calculates `logged_in` based on the value of `User.current_user`, instead of taking it as a parameter. But if `logged_in` were an optional parameter whose default value was set to `User.current_user.present?`, it'd work both for pre-existing calls, and for calculating `public_bookmarked_items_count` and `general_bookmarked_items_count`.)

Reply:
That makes sense. Are you asking me to modify the method in the way you're suggesting? I'm always a bit hesitant to modify existing code that may be used elsewhere, to avoid side affects, but with the reasonable default for the existing functionality here, this seems like it shouldn't be an issue, right?

Reply:
I don't think it'll cause any issues with the existing code, and I'd prefer not to have two different functions doing almost the same thing, so changing it seems like the best option to me.